src/Compilation/Config.zig

```diff
@@ -424,7 +424,10 @@ pub fn resolve(options: Options) ResolveError!Config {
     const debug_format: DebugFormat = b: {
         if (root_strip and !options.any_non_stripped) break :b .strip;
         break :b switch (target.ofmt) {
-            .elf, .macho, .wasm => .{ .dwarf = .@"32" },
+            .elf, .macho, .wasm => switch (options.debug_format.?.dwarf) {
+                .@"32" => .{ .dwarf = .@"32" },
+                .@"64" => .{ .dwarf = .@"64" },
+            },
```

Comment:
This works as expected with 

```
$ zig build-obj -gdwarf32 -O Debug -target x86_64-linux example.zig

$ /opt/homebrew/opt/binutils/bin/objdump --dwarf=info example.o | head -20

example.o:     file format elf64-x86-64

Contents of the .debug_info section:

  Compilation Unit @ offset 0:
   Length:        0x25f5a (32-bit)
   Version:       4
   Abbrev Offset: 0
   Pointer Size:  8
...
```

and 

```
$ zig build-obj -gdwarf64 -O Debug -target x86_64-linux example.zig

$ /opt/homebrew/opt/binutils/bin/objdump --dwarf=info example.o | head -20

example.o:     file format elf64-x86-64

Contents of the .debug_info section:

  Compilation Unit @ offset 0:
   Length:        0x31232 (64-bit)
   Version:       4
   Abbrev Offset: 0
   Pointer Size:  8
...
```

But, reviewing my code, I'm unsure of what the behavior would be if `debug_format` were empty. I can see that a 32-bit size is used when no flag is added:

```
$ zig build-obj -O Debug -target x86_64-linux example.zig

$ /opt/homebrew/opt/binutils/bin/objdump --dwarf=info example.o | head -20

example.o:     file format elf64-x86-64

Contents of the .debug_info section:

  Compilation Unit @ offset 0:
   Length:        0x25f5a (32-bit)
   Version:       4
   Abbrev Offset: 0
   Pointer Size:  8
...
```

Is this wanted?